Realm/Realm/Realm.cs

```diff
@@ -1448,29 +1430,32 @@ public void AddRealm(Realm realm)
             {
                 // We only want to have each realm once. That should be the case as AddRealm
                 // is only called in the Realm ctor, but let's check just in case.
-                Debug.Assert(!_weakRealms.Any(r =>
-                {
-                    return r.TryGetTarget(out var other) && ReferenceEquals(realm, other);
-                }), "Trying to add a duplicate Realm to the RealmState.");
+                Debug.Assert(!GetLiveRealms().Any(other => ReferenceEquals(realm, other)), "Trying to add a duplicate Realm to the RealmState.");
 
                 _weakRealms.Add(new WeakReference<Realm>(realm));
             }
 
-            public bool RemoveRealm(Realm realm)
+            /// <summary>
+            /// Remove a Realm from the list of Realms tracked by this state. This is only called when
+            /// Dispose is called on the Realm file and will not execute for garbage collected Realm
+            /// instances. This is fine because for GC-ed instances the lifecycle is as:
+            /// 1. Instance is GC-ed, its fields are GC-ed.
+            /// 2. The SharedRealmHandled is GC-ed, which causes Unbind to be called.
+            /// 3. The native pointer is deleted, which calls RealmCoordinator::unregister_realm.
+            /// 4. Once the last instance is deleted, the CSharpBindingContext destructor is called, which frees the state GCHandle.
+            /// 5. The State is now eligible for collection, and its fields will be GC-ed.
+            /// </summary>
+            public void RemoveRealm(Realm realm)
```

Comment:
According to the documentation this is called when `Dispose` is called:
- So the user does actually not call it themselves?
- Do we need it to be public then?

Reply:
The method is public, but the class it is defined in is internal:

https://github.com/realm/realm-dotnet/blob/4b316aaa1eb3b0341c7190269fdffacf73918c61/Realm/Realm/Realm.cs#L1415

So it won't be exposed to end users.